### PR TITLE
Require reCAPTCHA to create person records. #779

### PR DESCRIPTION
**Pull request by ryosuketc**
Require reCAPTCHA when creating person records to avoid fraud activities.
image

### CODE REVIEW

File: app/create.py [hunk 1]
```diff
@@ -385,11 +386,25 @@ def get(self):
...
         self.render('create.html',
+                    captcha_html=self.get_captcha_html(),
                     profile_websites=profile_websites,
                     profile_websites_json=simplejson.dumps(profile_websites),
                     onload_function='view_page_loaded')
 
     def post(self):
+        captcha_response = self.get_captcha_response()
+        if not captcha_response.is_valid:
+            self.params.create_mode = True
```
Comment (gimite) [inline app/create.py:397]: Can you factor out line 397-405 as a helper method as it is duplicate of get() method?

File: app/resources/create.html.template [hunk 2]
```diff
@@ -723,7 +723,7 @@
         </div>
 
         {% ifequal params.role "provide" %}
-          {% include "add_note_base.html.template" %}
+          {% include "add_note_base.html.template" with enable_recaptcha=True %}
```
Comment (gimite) [inline app/resources/create.html.template:726]: Can you handle the case where params.role is not "provide"? It looks Captcha is not shown in that case. This is the case when you first click "I'm looking for someone", search for some name and click [Create new record] at the bottom.

**Review by gimite — CHANGES_REQUESTED** [review]

File: app/create.py (hunk 1 above)
Comment (ryosuketc) [reply app/create.py]: Done.  _render_create_html.

File: app/resources/create.html.template (hunk 2 above)
Comment (ryosuketc) [reply app/resources/create.html.template:726]: Done. Added reCAPTCHA in the other branch (sorry my oversight).
image

**Comment by gimite** [comment]
Thanks! LGTM. Merged.

**Pull request merged**